lib/dependency-config.ts

```diff
@@ -0,0 +1,35 @@
+///<reference path=".d.ts"/>
+"use strict";
+
+import path = require("path");
+
+export class DependencyConfigService implements IDependencyConfigService {
+	private static DEPENDENCY_CONFIG_NAME = "dependency-config.json";
+
+	constructor(private $fs: IFileSystem) { }
+
+	public get dependencyConfigFilePath(): string {
+		return path.join(__dirname, "../config", DependencyConfigService.DEPENDENCY_CONFIG_NAME);
+	}
+
+	public getGeneratorConfig(generatorName: string): IFuture<IGeneratorConfig> {
+		return (() => {
+			var dependencyConfigContent = this.getDependencyConfigContent().wait();
+
+			var generatorConfig = _.find(dependencyConfigContent.generators, (generator: IGeneratorConfig) => generator.name === generatorName);
```

Comment:
```generatorName``` ultimately comes raw from the command line. The user can wRiTe LiKe ThIS. Should we massage the input before comparing it (maybe .toLowerCase())?